### PR TITLE
Video tracker incremental improvements

### PR DESCRIPTION
**Pull request by rpavlik**
Some incremental improvements that nevertheless improve tracking performance/quality.

### CODE REVIEW

File: plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp [hunk 1]
```diff
@@ -115,12 +117,16 @@ namespace vbtracker {
         m_permittedOutliers = permittedOutliers;
     }
 
-    bool BeaconBasedPoseEstimator::SetBeacons(const Point3Vector &beacons) {
-        return SetBeacons(beacons, DEFAULT_MEASUREMENT_VARIANCE);
+    bool BeaconBasedPoseEstimator::SetBeacons(
```
Comment (godbyk) [inline plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp]: Function names should generally start with a lowercase letter.

File: plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp [hunk 2]
```diff
@@ -115,12 +117,16 @@ namespace vbtracker {
...
+        const Point3Vector &beacons,
```
Comment (godbyk) [inline plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp]: `const` should follow typename (though I think clang-format will sort that out).

File: plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp [hunk 3]
```diff
@@ -115,12 +117,16 @@ namespace vbtracker {
...
     }
 
-    bool BeaconBasedPoseEstimator::SetBeacons(const Point3Vector &beacons,
-                                              double variance) {
+    bool BeaconBasedPoseEstimator::SetBeacons(
+        const Point3Vector &beacons, double variance,
+        BeaconIDPredicate const &autocalibrationFixedPredicate) {
         // Our existing pose won't match anymore.
         m_gotPose = false;
         m_beacons.clear();
```
Comment (godbyk) [inline plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp]: Call `m_beacons.reserve(beacons.size())`, too?

File: plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp [hunk 4]
```diff
@@ -166,13 +176,17 @@ namespace vbtracker {
...
-            beaconSum += cvToVector(beacon).cast<double>();
-            bNum++;
+        if (m_params.offsetToCentroid) {
+            Eigen::Vector3d beaconSum = Eigen::Vector3d::Zero();
+            auto bNum = size_t{0};
+            for (auto &beacon : beacons) {
+                beaconSum += cvToVector(beacon).cast<double>();
+                bNum++;
+            }
+            m_centroid = beaconSum / bNum;
```
Comment (godbyk) [inline plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp]: Prefer `std::accumulate() / beacons.size()`?

File: plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp [hunk 5]
```diff
@@ -447,9 +461,12 @@ namespace vbtracker {
                           out);
         return true;
     }
-
+#if 0
```
Comment (godbyk) [inline plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp]: Leave a comment explaining why this is commented out if you're going to leave it in the codebase.

File: plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp [hunk 6]
```diff
@@ -470,5 +487,14 @@ namespace vbtracker {
                   << m_state.getQuaternion().coeffs().transpose() << std::endl;
     }
 
+    void BeaconBasedPoseEstimator::dumpBeaconLocationsToStream(
```
Comment (godbyk) [inline plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp]: Isn't overriding `operator<<()` more idiomatic?

File: plugins/videobasedtracker/HDKLedIdentifierFactory.cpp [hunk 7]
```diff
@@ -169,6 +169,16 @@ namespace vbtracker {
         return ret;
     }
 
+    LedIdentifierPtr createHDKUnifiedLedIdentifier() {
+        LedIdentifierPtr ret;
+        std::vector<std::string> patterns = OsvrHdkLedIdentifier_SENSOR0_PATTERNS;
+        patterns.insert(end(patterns),
```
Comment (godbyk) [inline plugins/videobasedtracker/HDKLedIdentifierFactory.cpp]: Where is `begin()` and `end()` defined? Did you pull `std::begin()` into the current namespace? Some ADL magic?

File: plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp (hunk 1 above)
Comment (rpavlik) [inline plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp]: Not my doing.

File: plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp (hunk 2 above)
Comment (rpavlik) [inline plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp]: also not my doing.

File: plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp (hunk 4 above)
Comment (rpavlik) [inline plugins/videobasedtracker/BeaconBasedPoseEstimator.cpp]: will that work with the whole "have to convert opencv code, etc"?

File: plugins/videobasedtracker/HDKLedIdentifierFactory.cpp (hunk 7 above)
Comment (rpavlik) [inline plugins/videobasedtracker/HDKLedIdentifierFactory.cpp]: no, this is c++11 recommended thing, adl.

**Pull request merged**